myia/pipeline/steps.py

```diff
@@ -450,7 +467,11 @@ def wrapped(*args):
             raise MyiaInputTypeError('Wrong number of arguments.')
         args = tuple(_to_backend(to_canonical(arg, ot), backend, vt)
                      for arg, ot, vt in zip(args, orig_arg_t, vm_arg_t))
-        res = fn(*args)
+        if resources.universal:
+            unv, res = fn(new_universe, *args)
+            unv.commit()
```

Comment:
Why do you need a separate commit step here?

Reply:
To update all the states stored in the handles. So if e.g. you have a RNG, you will have a handle with its initial state, then Myia will keep track of the state using the universe, but nothing actually gets changed. `unv.commit()` will perform the update, so that when you call fn again, it will start where the first call left off.